Sources/SwiftParser/Patterns.swift

```diff
@@ -93,7 +102,10 @@ extension Parser {
       )
     case (.let, let handle)?,
       (.var, let handle)?,
-      (.inout, let handle)?:
+      (.inout, let handle)?,
+      (.mutating, let handle)?,
+      (.borrowing, let handle)?,
+      (.consuming, let handle)?:
```

Comment:
You don’t need to add all the cases to `ExpectedTokens` here, if you 
- Remove `let`, `var`, `inout` from `ExpectedTokens`
- Add a new `switch self.at(anyIn: ValueBindingPatternSyntax.BindingSpecifierOptions.self) {` here.

`ValueBindingPatternSyntax.BindingSpecifierOptions` is generated from the options you specify in `CodeGeneration`. It’s a recent-ish feature, that’s why it isn’t used yet. But a great opportunity to clean up the codebase a little.